### PR TITLE
Update PodReady condition when updating container readiness

### PR DESCRIPTION
**Pull request by timstclair**
This PR moves a few things around, but the new code is only what changed in `status.Manager.SetContainerReadiness` and the associated test (plus some code was moved into `status.Manager.updateStatusInternal` to make it common to all status updates).

Fixes https://github.com/kubernetes/kubernetes/issues/17656

### CODE REVIEW

**Comment by timstclair** [comment]
Sorry, forgot to push final version. Tests should be fixed now.

File: pkg/kubelet/status/manager.go [hunk 1]
```diff
@@ -205,16 +175,37 @@ func (m *manager) SetContainerReadiness(pod *api.Pod, containerID kubecontainer.
...
+		if condition.Type == api.PodReady {
+			readyConditionIndex = i
+			break
+		}
+	}
+	readyCondition := GeneratePodReadyCondition(&pod.Spec, status.ContainerStatuses, status.Phase)
+	if readyConditionIndex != -1 {
+		status.Conditions[readyConditionIndex] = readyCondition
+	} else {
+		glog.Warningf("PodStatus missing PodReady condition: %+v", status)
```
Comment (yujuhong) [inline pkg/kubelet/status/manager.go]: Why printing a warning? I assume a new pod would not have the ready condition.

Comment (timstclair) [inline pkg/kubelet/status/manager.go]: After reading through [kubelet.generatePodStatus](https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/kubelet.go#L3108), I don't think this should ever happen. I think in all the cases where there wouldn't be a ready condition, there also wouldn't be container statuses, so this function would abort earlier.

I can change it to a higher info level if you prefer though.

Comment (yujuhong) [inline pkg/kubelet/status/manager.go]: The logging level is fine if it's something that should not happen at all.

**Comment by yujuhong** [comment]
LGTM. Thanks for the nice cleanup.

**Comment by yujuhong** [comment]
@k8s-bot test this

**Comment by fabioy** [comment]
Manually merging to help test flakiness.

**Pull request merged**